src/main/java/com/myhexin/cragent/coderabbittest/AlarmChecker.java

```diff
@@ -0,0 +1,28 @@
+package com.myhexin.cragent.coderabbittest;
+
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class AlarmChecker {
+    
+    private static final ConcurrentHashMap<AlarmEvent, Long> EVENT_MAP = new ConcurrentHashMap<>();
+
+    public boolean needAlarm(AlarmModule module, String identity) {
+        AlarmEvent alarmEvent = new AlarmEvent(module, identity);
+        if (module.getExpiredSeconds() == 0) {
+            EVENT_MAP.put(alarmEvent, alarmEvent.createdNanoTime);
+            return true;
+        } else {
+            Long eventNanoTime = EVENT_MAP.compute(alarmEvent, (event, oldCreatedNanoTime) -> {
+                if (isExistedEventExpire(event, oldCreatedNanoTime)) {
+                    return event.createdNanoTime;
+                } else {
+                    return oldCreatedNanoTime;
+                }
+            });
+            return alarmEvent.createdNanoTime.equals(eventNanoTime);
+        }
+    }
```

Comment:
**Performance Issue**: Using a ConcurrentHashMap to store event times without considering the potential for very large datasets can lead to memory pressure and GC overhead, impacting the performance of the application. <br> **Fix**: Consider implementing an eviction policy for old or expired events from the EVENT_MAP to ensure the memory footprint is managed effectively. Alternatively, use a custom data structure optimized for the specific access patterns of your application. <br> **Code Suggestion**: 
 ```
 private static final ConcurrentHashMap<AlarmEvent, Long> EVENT_MAP = new CustomConcurrentHashMap<>();

 // CustomConcurrentHashMap should implement an eviction policy for old or expired events to manage memory effectively.
 ```